expression/collation.go

```diff
@@ -241,7 +241,7 @@ func deriveCollation(ctx sessionctx.Context, funcName string, args []Expression,
 		return ec, nil
 	case ast.Case:
 		// FIXME: case function aggregate collation is not correct.
-		return CheckAndDeriveCollationFromExprs(ctx, funcName, retType, args...)
+		return CheckAndDeriveCollationFromExprs(ctx, funcName, retType, args[1:]...)
```

Comment:
Why do we needn't consider args[0]? How about `select case when b < 1 then 1 else 1/0 end from t`?

Reply:
updated, we should only aggregate the `then expression`